app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb

```diff
@@ -62,12 +61,12 @@ def call_snapshot_delete
         _log.info("Deleting snapshot: reference: [#{mor}]")
         begin
           delete_snapshot(mor)
-        rescue => err
-          _log.error(err.to_s)
-          return
         rescue Timeout::Error
           msg = "Request to delete snapshot timed out"
           _log.error(msg)
```

Comment:
I'm wondering if the `rescue => err` was masking the fact that this is _not_ returning on a timeout error.  Following through it will then `set_status("Snapshot deleted: reference: [#{mor}]")` unless smartProxy and then `signal(:snapshot_complete)`.

That might be alright, the snap delete timed out but we're going to move on with the state machine.  Either way something to keep an eye on

Reply:
Though the fact that we're setting the status to `Snapshot deleted` and not `timed out` makes me think that path is expecting the delete to have been successful.

Reply:
I think I should probably add an explicit `return` on a timeout then.